org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/archetype/ArchetypeCatalogFactory.java

```diff
@@ -194,13 +189,31 @@ public ArchetypeCatalog getArchetypeCatalog() throws CoreException {
       return catalog;
     }
 
+    private String getLocalRepositoryURL() {
+      Path path;
+      try { // First try to use the id as a path, then as a URI else as it is
+        path = Path.of(getId());
+      } catch(Exception e1) {
+        try {
+          path = Path.of(new URI(getId()));
+        } catch(Exception e2) {
+          return getId();
+        }
+      }
+      path = path.toAbsolutePath();
+      if(Files.isRegularFile(path)) {
+        path = path.getParent();
+      }
+      return path.toUri().toString();
+    }
+
     private ArchetypeCatalog getEmbeddedCatalog() throws CoreException {
       URL url = getEmbeddedUrl();
       if(url == null) {
         //Not an embedded catalog, nothing else to do
         return null;
       }
-      try (InputStream is = new BufferedInputStream(url.openStream())) {
+      try (InputStream is = url.openStream()) {
```

Comment:
The ArchetypeCatalogXpp3Reader uses a buffered Stream/Reader already internally. Not need to buffer twice.